File: app/src/main/java/com/uet/uetworks/api/ApiBuilder.java
package com.uet.uetworks.api;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiBuilder {
    private static Api api;

    public static Api getApi(){
        if (api ==null){
            api = new Retrofit.Builder()
                    .baseUrl("http://192.168.1.121/chat1902e/")
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
                    .create(Api.class);
        }
        return api;
    }
}

File: app/src/main/java/com/uet/uetworks/api/ApiBuilder.kt
package com.uet.uetworks.api

import android.app.Service
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

internal object ApiBuilder {
    val BASE_URL = "http://112.137.129.69:8180"
    private var retrofit: Retrofit? = null

        val client: Retrofit?
            get() {
                val interceptor = HttpLoggingInterceptor()
                interceptor.level = HttpLoggingInterceptor.Level.BODY
                val client = OkHttpClient.Builder()
                    .addInterceptor(interceptor)
                    .build()


                if (retrofit == null) {
                    retrofit = Retrofit.Builder()
                        .baseUrl(BASE_URL)
                        .addConverterFactory(GsonConverterFactory.create())
                        .client(client)
                        .build()
                }
                return retrofit
            }
    }


